core/perf_test/test_sharedSpace.cpp

```diff
@@ -0,0 +1,300 @@
+
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#if defined _WIN32  // windows system
+#include <windows.h>
+unsigned getBytesPerPage() {
+  SYSTEM_INFO si;
+  GetSystemInfo(&si);
+  return si.dwPageSize;
+}
+#else  // unix/posix system
+#include <unistd.h>
+unsigned getBytesPerPage() { return sysconf(_SC_PAGESIZE); }
+#endif
+
+#include <Kokkos_Core.hpp>
+
+#include <algorithm>
+#include <numeric>
+#include <iostream>
+
+namespace {
+void printTimings(std::ostream& out, std::vector<double> const& tr,
+                  size_t numBytes, double threshold) {
+  out << "TimingResult contains " << tr.size() << " results:\n";
+  for (auto it = tr.begin(); it != tr.end(); ++it) {
+    out << "Duration of loop " << it - tr.begin() << " is " << *it
+        << " seconds.";
+    if ((*it) > threshold) {
+      out << " Marked as page migation. ";
+    }
+    out << "The transfer rate is " << numBytes / 1000000000 / (*it)
```

Comment:
Dumb question but should we use `numBytes / (1000)^3` or `numBytes / (1024)^3`?

Reply:
Whichever one is used, I think it's preferable for readability to write out the multiplication there, rather than writing the product directly. Or, if we stick with powers of 10, you can insert `'` as digit separators

Reply:
AFAIK `GB` is short for Gigabytes, thus `(1000)^3`, `GiB` is short for Gibibytes, thus `(1024)^3`. I think it is mostly a marketing gag for selling hard drives (that is acutally why a 16GB thumbdrive has less than you would expect). 

For something like a pci-e bus I think the most frequent thing is `giga bits per second` ... we could switch to that, but then the user has to think for himself.

For the output format: I will take whatever anyone prefers.

Reply:
I'm not a Kokkos developer, but here are some thoughts.

1. C++14 lets you use `'` as a digit separator for any decimal integer literal -- please do consider doing that.
2. Would you consider converting these numbers to `double` to avoid truncation when dividing?
3. `1000000000` is uncomfortably close to `numeric_limits<int>::max()`.  Would you consider expressing the literal as a `double` or as `1000000000LL` or `1000000000uLL`?

Reply:
@mhoemmen I switched to double and doing the literal as `pow(1000.0,3)` which I though is easiest to read together with a variable named `numBytes`.